widgetastic_manageiq/__init__.py

```diff
@@ -4878,3 +4878,55 @@ class MigrationDashboardStatusCard(AggregateStatusCard):
         ' h2[contains(@class,"card-pf-title")  and contains(.,{@name|quote})]]'
     )
     TITLE_ANCHOR = ".//h2"
+
+
+class AllAlertsList(Widget):
+    """Represents the list of fired alerts on the Monitor->All Alerts page"""
+
+    ROOT = ParametrizedLocator(".//div[contains(@class,{@list_class|quote})]")
+    ITEM_LOCATOR = './/div[contains(@class,"list-group-item alert")]'
+    ITEM_DESCRIPTION_LOCATOR = './/span[contains(@class,"description-column")]'
+    ITEM_EXPAND_LOCATOR = './/span[contains(@class,"fa-angle-right")]'
+    ITEM_PROVIDER_LOCATOR = './/a[@href="#"]/span[@class="no-wrap ng-binding"]'
+
+    def __init__(self, parent, list_class, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.list_class = list_class
+
+    @property
+    def all_items(self):
+        return [self.browser.text(item) for item in self.browser.elements(self.ITEM_LOCATOR)]
+
+    def read(self):
+        return self.all_items
+
+    def click_provider(self, provider_name):
+        provider_link = self._get_provider_link(provider_name)
+        self.browser.click(provider_link)
+
+    def click_item(self, item_description):
+        """Given a description, expands the list to display details for the alert.
```

Comment:
I would recommend implementing both `open_item` and `close_item` methods that click based on expansion state.

This way when using the page, I don't have to be concerned with checking the current expansion state, and accidentally opening instead of closing (or vice-versa). I can just call `view.alerts_list.open_item('alert 4')`